### PR TITLE
Fix: stream icon color in AllStreamList in dark theme. 

### PR DESCRIPTION
**Pull request by jainkuniya**
This PR is just rework of #2036, with smaller, separate commits :)
#2036 is quite old and at that time we weren't following this commit style. So I thought of created new PR instead of pushing into that.

* refactor: Rename local styles variable to componentStyles.   a75ba2a
So that styles from context can be de-structured and can be used
easily. Like the same way we do in other components.

* streamItem: Improve logic to determine icon color. 0969136
If color is not undefined then use it else decide it from
background. As if someone is explicitly specifying color
then it is expecting it to be used, rather than computing.
Fix: stream icon color is black in all stream screen in dark theme.

* flow: Add null & undefined params type to foregroundColorFromBackground. 382419d
Return 'white' for null and undefined color.

* refactor: Remove color from unreadView streamItem.  48f096a
As we are already sending stream color via backgroundColor
prop. And we want text, icon color to be decided from the
background color.
This removes the confusion in stream item that text and
icon color should be taken either from color or backgroundColor.

### CODE REVIEW

**Comment by borisyankov** [comment]
I think you haven't yet mastered the 'mutli-commit' PR.
Think of a PR as a 'single goal', and a commit as a 'single task'.

This PR can easily be 2 or 3 PRs. You don't need to group them so vaguely.
In addition to making stuff more complicated for yourself, you have not split the changes logically.

For example you change the logic in 0969136 but add the tests in 382419d

You are doing good job with doing smaller commits, now do the same with smaller PRs.

File: src/types.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ export type MapStateToProps = any; // { MapStateToProps } from 'react-redux';
 export type * from './actionTypes';
 export type * from './api/apiTypes';
 
+export type ColorValue = null | string | typeof undefined;
```
[the remark this answers is not in the document]
Comment (jainkuniya) [reply src/types.js]: RN are using it as `ColorValue` so I thought of keeping it same.
https://github.com/facebook/react-native/blob/d69e55060fd76d91eccc45905d250a9fce4b2c49/Libraries/StyleSheet/StyleSheetTypes.js#L16

> `export type Color = ?string;`

We are already importing `Color` from 'color', so I renamed it to `ColorType` in the same way like our other type names.

**Comment by jainkuniya** [comment]
Also broken down it to another 2 PR's (#2368 & #2367)

File: src/types.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (borisyankov) [reply src/types.js]: Then, ideally we should import that from RN and reuse.

[the remark this answers is not in the document]
Comment (jainkuniya) [reply src/types.js]: Done 👍

**Comment by borisyankov** [comment]
I think we need to simplify the color logic more, this is not ideal.
But good enough for now!

**Pull request merged**